### PR TITLE
HSEARCH-2986 Put the distribution files to upload into distribution/target/dir so that release scripts find them

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net//browse/HSEARCH-2986

### CODE REVIEW

File: distribution/src/main/assembly/dist.xml [hunk 1]
```diff
@@ -18,7 +18,10 @@
     <formats>
         <format>tar.gz</format>
         <format>zip</format>
+        <!-- Uncomment the following line to simplify content inspection: -->
+        <!--
```
Comment (Sanne) [inline distribution/src/main/assembly/dist.xml]: Thanks, I wanted to do this as well :)

**Comment by Sanne** [comment]
Looks good. The filter with "zip", that's on top of additional artifacts like the README, etc. right?

**Comment by yrodiere** [comment]
Yes, basically we put whatever files we want to upload into `distribution/target/dist`, and the release scripts will upload that plus the README etc.
See https://github.com/hibernate/hibernate-noorm-release-scripts/pull/4 for more information.
Maybe do not merge this PR right now, there seems to be some unexpected controversy on the changes I planned for the release scripts...

**Comment by yrodiere** [comment]
The release scripts have been updated, we're good to go! Merging...

**Pull request merged**